src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/ConnectionTestWithSSLCert/CertificateTest.cs

```diff
@@ -0,0 +1,245 @@
+        private const string LocalHost = "localhost";
+        private static readonly string s_fQDN = Dns.GetHostEntry(Environment.MachineName).HostName;
+        private readonly string _thumbprint;
+        private const string ForceEncryptionRegistryPath = @"SOFTWARE\Microsoft\Microsoft SQL Server\MSSQL16.MSSQLSERVER\MSSQLServer\SuperSocketNetLib";
+        private const string ThumbPrintEnvName = "Thumbprint";
+        #endregion
+
+        public CertificateTest()
+        {
+            CreateValidCertificate(s_fullPathToPowershellScript);
+            _thumbprint = Environment.GetEnvironmentVariable(ThumbPrintEnvName, EnvironmentVariableTarget.Machine);
+        }
+
+        [ConditionalFact(typeof(DataTestUtility), nameof(DataTestUtility.AreConnStringsSetup), nameof(DataTestUtility.IsManagedSNI), nameof(DataTestUtility.IsAdmin))]
+        [PlatformSpecific(TestPlatforms.Windows)]
+        public void OpenningConnectionWithGoodCertificateTest()
+        {
+            SqlConnectionStringBuilder builder = new(DataTestUtility.TCPConnectionString);
+
+            // confirm that ForceEncryption is enabled
+            using SqlConnection notEncryptedConnection = new(builder.ConnectionString);
+            builder.Encrypt = SqlConnectionEncryptOption.Optional;
+            notEncryptedConnection.Open();
+            Assert.Equal(ConnectionState.Open, notEncryptedConnection.State);
+
+            // Test with Mandatory encryption
+            builder.Encrypt = SqlConnectionEncryptOption.Mandatory;
+            using SqlConnection mandatoryConnection = new(builder.ConnectionString);
+            mandatoryConnection.Open();
+            Assert.Equal(ConnectionState.Open, mandatoryConnection.State);
+
+            // Test with strict encryption
+            builder.Encrypt = SqlConnectionEncryptOption.Strict;
+            using SqlConnection strictConnection = new(builder.ConnectionString);
+            strictConnection.Open();
+            Assert.Equal(ConnectionState.Open, strictConnection.State);
+        }
+
+        // Provided hostname in certificate are:
+        // localhost, FQDN, Loopback IPv4: 127.0.0.1, IPv6: ::1
+        [ConditionalFact(typeof(DataTestUtility), nameof(DataTestUtility.AreConnStringsSetup), nameof(DataTestUtility.IsManagedSNI), nameof(DataTestUtility.IsAdmin))]
+        [PlatformSpecific(TestPlatforms.Windows)]
+        public void OpeningConnectionWitHNICTest()
+        {
+            // Mandatory
+            SqlConnectionStringBuilder builder = new(DataTestUtility.TCPConnectionString)
+            {
+                // 127.0.0.1 most of the cases does not cause any Remote certificate validation error depinding on name resolution on the machine
+                //  It mostly returns SslPolicyErrors.None
+                DataSource = IPV4,
+                Encrypt = SqlConnectionEncryptOption.Mandatory,
+                HostNameInCertificate = "localhost"
```

Comment:
Can you check whether this works without specifying this in Native SNI?
Also do run all tests with both Managed and Native SNI whenever possible, and behavior should be identical.